cirq-core/cirq/sim/act_on_density_matrix_args.py

```diff
@@ -68,8 +68,16 @@ def __init__(
         """
         super().__init__(prng, qubits, log_of_measurement_results, ignore_measurement_results)
         self.target_tensor = target_tensor
-        self.available_buffer = available_buffer
-        self.qid_shape = qid_shape
+        if available_buffer is None:
+            self.available_buffer = [np.empty_like(target_tensor) for _ in range(3)]
+        else:
+            self.available_buffer = available_buffer
+        if qid_shape is None:
+            target_shape = target_tensor.shape
+            assert len(target_shape) % 2 == 0
```

Comment:
This should `raise ValueError`. (Typically users wouldn't call this constructor directly so assertion kind of makes sense here, but it's still a public function of a public class, so raising explicit errors is preferred).

Reply:
Agree. It has been updated in the new commit. A test case is added as well.